### DISCUSSION TITLE
p-dropdown's showClear is not working if a p-calendar is present

### DISCUSSION THREAD

**Discussion by danieljanosrobert** [primefaces/community#1012] [PrimeNG] [answered]
Hello!
I need a p-dropdown where I already have a p-calendar. I just updated to 17.3.3 and the showClear has stopped working.
You can reproduce this in the following dropdown reactive forms demo: https://stackblitz.com/edit/jjpzwu?file=src%2Fapp%2Fdemo%2Fdropdown-reactive-forms-demo.html,package.json

```html
<div class="card flex justify-content-center">
  <form [formGroup]="formGroup">
    <p-dropdown
      formControlName="selectedCity"
      [showClear]="true"
      [options]="cities"
      optionLabel="name"
      placeholder="Select a City"
    ></p-dropdown>
    <p-calendar></p-calendar>
  </form>
</div>
```

The times icon is present, but it does not clear the content, instead it just opens the dropdown.
It works as intended without the calendar.

Edit: In 17.3.2 it worked well with the calendar.

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

This issue has been fixed and the fix will be available in the upcoming release next week.
https://github.com/primefaces/primeng/issues/14606

**Comment by bhupendra-99** [comment]
when using [showIcon]="true" and [showClear]="true" they overlap each other on p-calendar

```
<p-calendar
              [appendTo]="'body'"
              [showIcon]="true"
              [showClear]="true"
              pInputDate
              id="creationDateEnd"
              name="creationDateEnd"
              [dateFormat]="calendarDateFormat"
              formControlName="creationDateEnd"
              [readonlyInput]="true"
              [(ngModel)]="vm.creationEndDate"
              [pTooltip]="vm.creationEndDate"
              tooltipPosition="bottom"
            >
</p-calendar>
```
